packages/combo-box/src/vaadin-combo-box-mixin.js

```diff
@@ -930,6 +930,14 @@ export const ComboBoxMixin = (subclass) =>
           this.opened || this.autoOpenDisabled
             ? this.$.dropdown.indexOfLabel(this.filter)
             : this._indexOfValue(this.value, this.filteredItems);
+
+        // see https://github.com/vaadin/web-components/issues/2615
+        if (this.value && this._focusedIndex >= 0) {
+          const filteredValue = this.filteredItems[this._focusedIndex];
```

Comment:
nit: the variable name should probably be `filteredItem`. The _value_ for it is obtained in the next row.

Reply:
Done.